services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -109,28 +118,76 @@ public static Builder builder() {
   public abstract String language();
 
   /**
-   * The same radiuses were used during the request that resulted in this root directions response.
+   * The maximum distance a coordinate can be moved to snap to the road network in meters. There
+   * must be as many radiuses as there are coordinates in the request, each separated by ;.
+   * Values can be any number greater than 0 or the string unlimited. A NoSegment error is returned
+   * if no routable road is found within the radius.
    *
-   * @return a string representing the radiuses
+   * @return a string representing the radiuses separated by ;.
    * @since 3.0.0
+   * @deprecated use {@link #radiusesList()} ()}
+   */
+  @Nullable
+  @Deprecated
+  public String radiuses() {
+    return FormatUtils.formatRadiuses(radiusesList());
+  }
+
+  /**
+   * The maximum distance a coordinate can be moved to snap to the road network in meters. There
+   * must be as many radiuses as there are coordinates in the request.
+   * Values can be any number greater than 0 or the string unlimited. A NoSegment error is returned
```

Comment:
The "unlimited" string is not going to be returned here. What happens when we supply "unlimited" or "NoSegment" to `Double.valueOf`?

Reply:
`println(Double.POSITIVE_INFINITY)` -> `Infinity`
`print("unlimited".toDouble())` -> `NumberFormatException`
`print("NoSegment".toDouble())` -> `NumberFormatException`